frontend/src/common/ReviewWithComment.tsx

```diff
@@ -60,6 +62,7 @@ export default function ReviewWithComment({
 
   const { responses, isResponsesLoading, isResponsesError } = useGetResponses([...reviews.map((review) => review._id)])
   const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles(modelId)
+  const { role } = router.query
   const [reviewRequest, setReviewRequest] = useState(reviews[0])
```

Comment:
You can do all of this a little easier by just changing this line to
```
  const [reviewRequest, setReviewRequest] = useState<ReviewRequestInterface>(
    reviews.find((review) => review.role === role) || reviews[0],
  )
```
I've added in an explicit type for the `useState` as TS seems to be losing track of it.